plugins/generator-1.18.1/forge-1.18.1/templates/fluid/fluidgen.java.ftl

```diff
@@ -0,0 +1,97 @@
+<#--
+ # MCreator (https://mcreator.net/)
+ # Copyright (C) 2012-2020, Pylo
+ # Copyright (C) 2020-2021, Pylo, opensource contributors
+ #
+ # This program is free software: you can redistribute it and/or modify
+ # it under the terms of the GNU General Public License as published by
+ # the Free Software Foundation, either version 3 of the License, or
+ # (at your option) any later version.
+ #
+ # This program is distributed in the hope that it will be useful,
+ # but WITHOUT ANY WARRANTY; without even the implied warranty of
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+ #
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ #
+ # Additional permission for code generator templates (*.ftl files)
+ #
+ # As a special exception, you may create a larger work that contains part or
+ # all of the MCreator code generator templates (*.ftl files) and distribute
+ # that work under terms of your choice, so long as that work isn't itself a
+ # template for code generation. Alternatively, if you modify or redistribute
+ # the template itself, you may (at your option) remove this special exception,
+ # which will cause the template and the resulting code generator output files
+ # to be licensed under the GNU General Public License without this special
+ # exception.
+-->
+
+<#-- @formatter:off -->
+<#include "../mcitems.ftl">
+<#include "../procedures.java.ftl">
+
+package ${package}.world.features.lakes;
+
+import net.minecraft.world.level.levelgen.feature.stateproviders.BlockStateProvider;
+
+public class ${name}Feature extends LakeFeature {
+	public static final ${name}Feature FEATURE = (${name}Feature) new ${name}Feature().setRegistryName("${modid}:${registryname}");
+	public static final ConfiguredFeature<?, ?> CONFIGURED_FEATURE = FEATURE.configured(new LakeFeature.Configuration(
+			BlockStateProvider.simple(${JavaModName}Blocks.${data.getModElement().getRegistryNameUpper()}), BlockStateProvider.simple(Blocks.AIR)));
+	public static final PlacedFeature PLACED_FEATURE = CONFIGURED_FEATURE.placed(List.of(
+			RarityFilter.onAverageOnceEvery(${data.frequencyOnChunks}),
+			InSquarePlacement.spread(),
+			PlacementUtils.HEIGHTMAP

```

Comment:
Something I noticed with this placement is that now lakes can't generate underground anymore. Maybe `PlacementUtils.RANGE_BOTTOM_TO_MAX_TERRAIN_HEIGHT` would be a better choice (from bottom of the world to 256). 
Until we move to a better system, world gen should remain as close as possible to the older versions

Reply:
I'll try that ASAP

Reply:
At this point I wanna note that at first I went with `PlacementUtils.FULL_RANGE` and lakes failed to generate. This gives me a vibe that `VerticalAnchor.top()` is assigned to way higher height level than 256.

Reply:
Because it is - at least for the Overworld. It's at 320 now, but since the world only generates up to 256, you had 64 impossible heights

Reply:
Doesn't work - NO LAKES: neither surface, nor underground.
Lava lakes use separate placement configurations for them, are you keen on this approach?

Reply:
The lakes do generate, both on surface and underground
![on surface](https://user-images.githubusercontent.com/64546452/154137354-703131a1-9b35-4e79-9b79-9e7487284f49.png)
![underground](https://user-images.githubusercontent.com/64546452/154137432-f3beac49-26d5-4ac4-9a36-825da2805b98.png)

Reply:
Weird. I didn't see that :confused:

Reply:
:heavy_check_mark: Fixed